src/routers/alpha-router/gas-models/mixedRoute/mixed-route-heuristic-gas-model.ts

```diff
@@ -137,6 +94,44 @@ export class MixedRouteHeuristicGasModelFactory extends IOnChainGasModelFactory
         providerConfig
       );
 
+      /** ------ MARK: USD Logic -------- */
+      const gasCostInTermsOfUSD = getQuoteThroughNativePool(
+        chainId,
+        totalGasCostNativeCurrency,
+        usdPool
+      );
+
+      /** ------ MARK: Conditional logic run if gasToken is specified  -------- */
+      const nativeGasTokenPool: Pool | null = pools.nativeGasTokenV3Pool;
+      let gasCostInTermsOfGasToken: CurrencyAmount | undefined = undefined;
+      if (nativeGasTokenPool) {
+        gasCostInTermsOfGasToken = getQuoteThroughNativePool(
+          chainId,
+          totalGasCostNativeCurrency,
+          nativeGasTokenPool
+        );
+      }
+      // if the gasToken is the native currency, we can just use the totalGasCostNativeCurrency
+      else if (providerConfig?.gasToken?.equals(nativeCurrency)) {
+        gasCostInTermsOfGasToken = totalGasCostNativeCurrency;
+      }
+
+      /** ------ MARK: return early if quoteToken is wrapped native currency ------- */
+      if (quoteToken.equals(nativeCurrency)) {
```

Comment:
nit: it seems like if `quoteToken.equals(nativeCurrency)` is true, then `providerConfig?.gasToken?.equals(nativeCurrency)` is also true, meaning we can even return earlier.

But I do think this way the readability degrades a bit.

Reply:
mmm no so the gasToken is not always the quoteToken, in fact the main use case of this is to allow for us to quote a route's estimated gas in a third token that is not the tokenIn or tokenOut